msal/oauth2cli/default_http_client.py

```diff
@@ -0,0 +1,46 @@
+import requests
+
+from .http import HttpClient, Response
+
+
+class DefaultHttpClient(HttpClient):
```

Comment:
I understand that there has been multiple efforts on fixing issues in this implementation, like [this initial attempt](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/169/commits/b3a4e09a3cb845cb0244143228a3bc60d405b3fd#diff-102c5027a0791530c6636cd879afe485R46), [this fix](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/169/commits/ccafcf96a15f8d2e9d43dc7d27a3f12cca9125df#diff-102c5027a0791530c6636cd879afe485R44), and the [latest change](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/169/commits/2670e254511a58f6f6b104355a80a21672f4893c#diff-102c5027a0791530c6636cd879afe485R35) (whose interface may or may not be better than the previous one). With all these understanding, let's continue our previous conversation on whether we need to implement and then carry the burden of maintaining such a DefaultHttpClient. Do we really want/need to maintain a DefaultHttpClient, if we do not have to?

In other words, historically msal python uses requests as the underlying http layer, now this task is to "allow customizable http layer (when the customer chooses to do so)". That's it. This task does not imply we have to swap the default http from the battle-tested requests to a home-made http layer.

This DefaultHttpClient can be moved into the test infrastructure, though. It would be useful there.

Reply:
This implementation still uses the battle-tested requests. It only separates it out to its own layer instead of having it called as and when needed. One of the things which was very confusing when trying to do this task was to actually make sure all the instances of requests are given a functionality of customization. Now you wont touch this layer as it is one of the inner layers similar to oauth2 unless it is needed, but if changes related to requests are needed we will have a single place to make them